frontend/src/static/js/v_summary.js

```diff
@@ -323,42 +377,76 @@ window.vSummary = {
         isSortingWithinDsc: this.isSortingWithinDsc,
       };
       this.filtered = this.sortFiltered(this.filtered, filterControl);
+    },
 
-      if (this.isMergeGroup) {
-        this.mergeGroup(this.filtered);
-      }
+    getMergedRepos() {
+      this.filtered.forEach((group, groupIndex) => {
+        if (this.mergedGroupsMap[this.getGroupName(group)]) {
+          this.mergeGroupByIndex(this.filtered, groupIndex);
+        }
+      });
     },
 
-    mergeGroup(filtered) {
-      filtered.forEach((group, groupIndex) => {
-        const dateToIndexMap = {};
-        const mergedCommits = [];
-        const mergedFileTypeContribution = {};
-        let mergedVariance = 0;
-        let totalMergedCommits = 0;
-        let totalMergedCheckedFileTypeCommits = 0;
+    mergeGroupByIndex(filtered, groupIndex) {
+      const dateToIndexMap = {};
+      const mergedCommits = [];
+      const mergedFileTypeContribution = {};
+      let mergedVariance = 0;
+      let totalMergedCommits = 0;
+      let totalMergedCheckedFileTypeCommits = 0;
 
-        group.forEach((user) => {
-          this.mergeCommits(user, mergedCommits, dateToIndexMap);
+      filtered[groupIndex].forEach((user) => {
+        this.mergeCommits(user, mergedCommits, dateToIndexMap);
 
-          this.mergeFileTypeContribution(user, mergedFileTypeContribution);
+        this.mergeFileTypeContribution(user, mergedFileTypeContribution);
 
-          totalMergedCommits += user.totalCommits;
-          totalMergedCheckedFileTypeCommits += user.checkedFileTypeContribution;
-          mergedVariance += user.variance;
-        });
+        totalMergedCommits += user.totalCommits;
+        totalMergedCheckedFileTypeCommits += user.checkedFileTypeContribution;
+        mergedVariance += user.variance;
+      });
+
+      mergedCommits.sort(window.comparator((ele) => ele.date));
+      filtered[groupIndex][0].commits = mergedCommits;
+      filtered[groupIndex][0].fileTypeContribution = mergedFileTypeContribution;
+      filtered[groupIndex][0].totalCommits = totalMergedCommits;
+      filtered[groupIndex][0].variance = mergedVariance;
+      filtered[groupIndex][0].checkedFileTypeContribution = totalMergedCheckedFileTypeCommits;
+
+      filtered[groupIndex] = filtered[groupIndex].slice(0, 1);
```

Comment:
can place a comment here to explain why only the first subarray is taken. (aka the merged group only)